### PR TITLE
Migrated helpLinkTextInput to ui-shared

### LINKED ISSUE #27769: Migrate HelpLinkTextInput to ui-shared
### Description

Migrate HelpLinkTextInput to ui-shared

### PR DESCRIPTION
**Pull request by agagancarczyk**
Closes https://github.com/keycloak/keycloak/issues/27769

### CODE REVIEW

File: js/libs/ui-shared/src/controls/TextControl.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export type TextControlProps<
   P extends FieldPath<T> = FieldPath<T>,
 > = UseControllerProps<T, P> &
   Omit<TextInputProps, "name" | "isRequired" | "required"> & {
-    label: string;
+    label?: string;
```
Comment (jonkoops) [inline js/libs/ui-shared/src/controls/TextControl.tsx:18]: A `label` must be specified at all times, so let's not make this optional here.

File: js/apps/admin-ui/src/realm-settings/security-defences/HelpLinkTextInput.tsx [hunk 2]
```diff
@@ -16,25 +14,27 @@ export const HelpLinkTextInput = ({
...
+              <Trans i18nKey={`${name}Help`}>
+                Default value prevents pages from being included
+                <FormattedLink href={url} title={t("learnMore")} />
+              </Trans>
+            }
+            fieldLabelId={name}
+          />
+        }
+      >
+        <TextControl name={fieldName} />
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/security-defences/HelpLinkTextInput.tsx:36]: A `TextControl` already includes a `FormGroup`, so I don't think this is the right use-case. Either we replace the while `FormGroup` above, or we leave this unported and handle it manually in the PF5 upgrade.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/realm-settings/security-defences/HelpLinkTextInput.tsx (hunk 2 above)
Comment (agagancarczyk) [reply js/apps/admin-ui/src/realm-settings/security-defences/HelpLinkTextInput.tsx:36]: Thanks @jonkoops maybe let's leave it and handle it manually in the PF5 upgrade.

Comment (jonkoops) [reply js/apps/admin-ui/src/realm-settings/security-defences/HelpLinkTextInput.tsx:36]: Sure!

**Pull request closed without merging**